tests/Unit/Domain.LinnApps.Tests/PartsReceivedReportServiceTests/WhenGettingReport.cs

```diff
@@ -0,0 +1,91 @@
+                                        MaterialPrice = 11,
+                                        DateBooked = DateTime.UnixEpoch,
+                                        OrderNumber = "100 / 1",
+                                        JobRef = "AAA",
+                                        OverStockValue = 1,
+                                        OverstockQty = 1,
+                                        PartPrice = 1,
+                                        TqmsGroup = "LINN",
+                                        Qty = 1
+                                    },
+                                new PartsReceivedViewModel
+                                    {
+                                        SupplierName = "SUPPLIER 2",
+                                        PartNumber = "PART 2",
+                                        MaterialPrice = 22,
+                                        DateBooked = DateTime.UnixEpoch,
+                                        OrderNumber = "200 / 1",
+                                        JobRef = "AAA",
+                                        OverStockValue = 2,
+                                        OverstockQty = 2,
+                                        PartPrice = 2,
+                                        TqmsGroup = "LINN",
+                                        Qty = 2
+                                    },
+                                new PartsReceivedViewModel
+                                    {
+                                        SupplierName = "SUPPLIER 3",
+                                        PartNumber = "PART 3",
+                                        MaterialPrice = 33,
+                                        DateBooked = DateTime.UnixEpoch.AddDays(1),
+                                        OrderNumber = "300 / 1",
+                                        JobRef = "AAA",
+                                        OverStockValue = 3,
+                                        OverstockQty = 3,
+                                        PartPrice = 3,
+                                        TqmsGroup = "LINN",
+                                        Qty = 3
+                                    }
+                            };
+            this.PartsReceivedView.FilterBy(Arg.Any<Expression<Func<PartsReceivedViewModel, bool>>>())
+                .Returns(this.data.AsQueryable());
+
+            this.results = this.Sut.GetReport(
+                "AAA",
+                null,
+                DateTime.UnixEpoch.ToString("o"),
+                DateTime.Today.ToString("o"),
+                false,
+                string.Empty);
+        }
+
+        [Test]
+        public void ShouldReturnData()
+        {
+            this.results.ReportTitle.DisplayValue.Should().Be(
+                $"Parts Received between {DateTime.UnixEpoch.ToShortDateString()} and {DateTime.Today.ToShortDateString()}");
+            this.results.Rows.Count().Should().Be(3);
```

Comment:
There's the nice GetGridTextValue syntax if you do want to check any of the data's in the right bit but maybe overkill as there's not too much in the domain service.. guess it could cover the negative values filter or order by if you wanted

Reply:
Yeah, I ended up writing tests to check all the filters and order bys were applied correctly in the end. Turns out they weren't and I had to make some fixes. Hurray for tests!